src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java

```diff
@@ -434,8 +434,12 @@ public Mono<BookingResponseTO> updateBookingByReferenceCarrierBookingRequestRefe
                 // resolve entities linked to booking
                 createDeepObjectsForBooking(bookingRequest, booking)
                     .flatMap(
-                        bTO ->
-                            createShipmentEventFromBookingTO(booking.getId(), bTO).thenReturn(bTO)))
+                        bookingTO ->
+                            Mono.just(
+                                    shipmentEventMapper.shipmentEventFromBookingTO(
+                                        bookingTO, booking.getId(), null))
+                                .flatMap(this::createShipmentEvent)
```

Comment:
```suggestion
                            this.createShipmentEvent(shipmentEventMapper.shipmentEventFromBookingTO(
                                        bookingTO, booking.getId(), null))
```
(possibly with a different layout - but in any event, the `Mono.just` is redundant.

Reply:
changed accordingly